Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -207,6 +209,25 @@ def check_python_packages(lib, deps)
     EOS
   end
 
+  def check_repository_references(prefix)
```

Comment:
Should/could this be specific to cases where `HOMEBREW_PREFIX == HOMEBREW_REPOSITORY`? It doesn't seem like it'd be a problem elsewhere.

Reply:
I can't at the moment think of a file under `$HOME_REPOSITORY` that is not also under `$HOMEBREW_LIBRARY` that should ever be referenced in a bottle. The other folders under `$HOMBREW_REPOSITORY` are `bin`, `completions`, `docs`, and `manpages`. I'd be inclined to leave this audit enabled unconditionally. We could revisit if we were to find a valid counterexample.

Reply:
I think on audits we should be cautious and avoid adding audits that will fail in configurations where they don't actually matter.

```suggestion
  def check_repository_references(prefix)
    return if HOMEBREW_PREFIX != HOMEBREW_REPOSITORY
```

Reply:
Additionally, thinking more about this: if this was a hard failure in `brew bottle` rather than a `brew audit` failure then it would allow simplifying https://github.com/Homebrew/brew/blob/2be340c6d6495774c891fbd2d3af37614d3c89d0/Library/Homebrew/software_spec.rb#L381-L387 and https://github.com/Homebrew/brew/blob/2be340c6d6495774c891fbd2d3af37614d3c89d0/Library/Homebrew/formula_installer.rb#L177-L187 to ignore the repository entirely in every case.